File: 백준/스택의 활용/송예은/쇠막대기.py
string = list(input())
stack = []
answer = 0

for i in range(len(string)):
  if string[i] == '(':
    stack.append('(')
  else :
    if stack and string[i-1] == '(':
      stack.pop()
      answer += len(stack)
    else :
      stack.pop()
      answer += 1
print(answer)

File: 백준/스택의 활용/문종후/쇠막대기.js
const fs = require("fs");
const input = fs.readFileSync("example.txt").toString();
let stack = [];
let answer = 0;
let n = input.length;
for (let i = 0; i < n; i++) {
  if (input[i] === "(") {
    stack.push(input[i]);
  } else {
    stack.pop();
    if (input[i - 1] == "(") {
      answer += stack.length;
    } else {
      answer++;
    }
  }
}

console.log(answer);
